epochStart/metachain/rewards.go

```diff
@@ -59,7 +59,7 @@ func (rc *rewardsCreator) CreateRewardsMiniBlocks(
 	defer rc.mutRewardsData.Unlock()
 
 	rc.clean()
-	rc.flagDelegationSystemSCEnabled.SetValue(metaBlock.GetEpoch() >= rc.delegationSystemSCEnableEpoch)
+	rc.flagDelegationSystemSCEnabled.SetValue(metaBlock.GetEpoch() >= rc.enableEpochsHandler.StakingV2EnableEpoch())
```

Comment:
why do we still need this flag?

Reply:
kept the flag as it is set based on the epoch from metablock, not the epoch received on `EpochConfirmed`